lib/taskana-core/src/main/java/pro/taskana/TaskanaEngineConfiguration.java

```diff
@@ -489,31 +283,415 @@ public static boolean shouldUseLowerCaseForAccessIds() {
+      this.corpusChristiEnabled = tec.isCorpusChristiEnabled();
+      this.jobBatchSize = tec.getMaxNumberOfUpdatesPerTransaction();
+      this.maxNumberOfJobRetries = tec.getMaxNumberOfJobRetries();
+      this.cleanupJobFirstRun = tec.getCleanupJobFirstRun();
+      this.cleanupJobRunEvery = tec.getCleanupJobRunEvery();
+      this.cleanupJobMinimumAge = tec.getCleanupJobMinimumAge();
+      this.taskCleanupJobAllCompletedSameParentBusiness =
+          tec.isTaskCleanupJobAllCompletedSameParentBusiness();
+      this.validationAllowTimestampServiceLevelMismatch =
+          tec.isValidationAllowTimestampServiceLevelMismatch();
+      this.addAdditionalUserInfo = tec.isAddAdditionalUserInfo();
+      this.priorityJobBatchSize = tec.getPriorityJobBatchSize();
+      this.priorityJobFirstRun = tec.getPriorityJobFirstRun();
+      this.priorityJobRunEvery = tec.getPriorityJobRunEvery();
+      this.priorityJobActive = tec.isPriorityJobActive();
+      this.userRefreshJobRunEvery = tec.getUserRefreshJobRunEvery();
+      this.userRefreshJobFirstRun = tec.getUserRefreshJobFirstRun();
+      this.minimalPermissionsToAssignDomains = tec.getMinimalPermissionsToAssignDomains();
     }
 
-    try (Connection connection = dataSource.getConnection()) {
-      String databaseProductId =
-          DB.getDatabaseProductId(connection.getMetaData().getDatabaseProductName());
-      if (DB.isPostgres(databaseProductId)) {
-        this.schemaName = this.schemaName.toLowerCase();
+    public TaskanaEngineConfigurationBuilder(
+        DataSource dataSource, boolean useManagedTransactions, String schemaName) {
+      this(dataSource, useManagedTransactions, true, schemaName);
+    }
+
+    public TaskanaEngineConfigurationBuilder(
+        DataSource dataSource,
+        boolean useManagedTransactions,
+        boolean securityEnabled,
+        String schemaName) {
+      this(dataSource, useManagedTransactions, securityEnabled, schemaName, null, null);
+    }
+
+    public TaskanaEngineConfigurationBuilder(
+        DataSource dataSource,
+        boolean useManagedTransactions,
+        boolean securityEnabled,
+        String schemaName,
+        String propertiesFileName,
+        String propertySeparator) {
+      this.useManagedTransactions = useManagedTransactions;
+      this.securityEnabled = securityEnabled;
+
+      if (dataSource != null) {
+        this.dataSource = dataSource;
       } else {
-        this.schemaName = this.schemaName.toUpperCase();
+        // use default In Memory datasource
+        this.dataSource = createDefaultDataSource();
       }
-    } catch (SQLException ex) {
-      LOGGER.error("Caught exception when attempting to initialize the schema name", ex);
+
+      this.schemaName = initSchemaName(schemaName);
+
+      this.initTaskanaProperties(
+          propertiesFileName == null ? TASKANA_PROPERTIES : propertiesFileName,
+          propertySeparator == null ? TASKANA_PROPERTY_SEPARATOR : propertySeparator);
```

Comment:
not sure if this is relevant (due to #2084). Replace the null check with the correct defaults in the overloaded constructors?